include/openssl/rsa.h

```diff
@@ -215,6 +215,88 @@ OPENSSL_EXPORT int RSA_set0_factors(RSA *rsa, BIGNUM *p, BIGNUM *q);
 OPENSSL_EXPORT int RSA_set0_crt_params(RSA *rsa, BIGNUM *dmp1, BIGNUM *dmq1,
                                        BIGNUM *iqmp);
 
+// RSA_METHOD functions
+
+// RSA_get_default_method returns a zero initialized |RSA_METHOD| object. The
+// wrapper functions will select the appropriate |rsa_default_*| implementation.
+OPENSSL_EXPORT const RSA_METHOD *RSA_get_default_method(void);
+
+// RSA_meth_new returns a zero-initialized |RSA_METHOD| object. It sets
+// |flags| on the object. The |name| parameter is currently ignored and
+// part of the function signature for OpenSSL compatibility.
+OPENSSL_EXPORT RSA_METHOD *RSA_meth_new(const char *name, int flags);
+
+// RSA_set_method sets |meth| on |rsa|. Returns one on success and zero
+// on failure.
+OPENSSL_EXPORT int RSA_set_method(RSA *rsa, const RSA_METHOD *meth);
+
+// RSA_get_method returns the |RSA_METHOD| object associated with |rsa|.
+OPENSSL_EXPORT const RSA_METHOD *RSA_get_method(const RSA *rsa);
+
+// RSA_meth_free frees the memory associated with |meth|
+OPENSSL_EXPORT void RSA_meth_free(RSA_METHOD *meth);
+
+// RSA_METHOD setters
+// The following functions set the corresponding fields on |meth|. Returns one
+// on success and zero on failure.
+
+// RSA_meth_set_init sets |init| on |meth|. |init| should return one on
+// success and zero on failure.
+OPENSSL_EXPORT int RSA_meth_set_init(RSA_METHOD *meth, int (*init) (RSA *rsa));
+
+// RSA_meth_set_finish sets |finish| on |meth|. The |finish| function
+// is called in |RSA_free| before freeing the key. |finish| should return
+// one on success and zero on failure.
+OPENSSL_EXPORT int RSA_meth_set_finish(RSA_METHOD *meth,
+                                       int (*finish) (RSA *rsa));
+
+// RSA_meth_set_priv_dec sets |priv_dec| on |meth|. The |priv_dec| function
+// should return the number of bytes written to the object |to| or -1 for error.
+// |priv_dec| should decrypt |max_out| bytes at |from| using the private key
+// |rsa| and store the plaintext in |to|. |priv_dec| should return the size of
+// the recovered plaintext or a negative number on error.
```

Comment:
No we will not need to designate another negative number. Generally OpenSSL uses -1 for error. Since this if for the custom implementations, we can be more generic and say a negative number indicates error/. Our checks align with this (we check if return value is less than 0). This would encompass the -1 case as well.

Reply:
Do we want to shorten the documentation and avoid duplication here and in the next 3 as suggested above?

Reply:
Only the second sentence is duplicated across the comments. There are slight differences otherwise in what each func should be doing. I think it's okay to leave it like this, might make it more confusing to have two separate documentation sections for these funcs.